### ISSUE TITLE
model.evaluate() in Keras do not cover all datapoints

### ISSUE THREAD

**Issue by thepunitsingh** [fchollet/keras#6889]
I am running `theano.__version__ = 0.9.0.dev-c697eeab84e5b8a74908da654b66ec9eca4f1291` and `keras.__version__ = 2.0.2` on conda environment with Python 3.6.1. OS in Windows Server 2012 R2 with Graphics Card Nvidia Quadro K6000. Cuda 7.5 installed with Nvidia Driver 354.56.

`model.evaluate()` is not covering all data points. I have total 768 data points, but `model.evaluate()` is evaluating only 32 data points and give the accuracy of 75.52%. I also tried batch sizes of 100, 50, 20, 10 & 1, but it does not cover all the data points but the accuracy remains unchanged.

You can check my Jupyter Notebook file if the implementation [here](https://github.com/thepunitsingh/keras_tutorials/blob/master/pima-indians-diabetes-categorization.ipynb)

I tried to discuss this on [StackExchange](https://stackoverflow.com/q/44384924/6701627), and I was suggested by experience people that it could be a bug in keras or the progress bar might have broken. I don't know how to check if the progress bar is broken, so I am posting the issue here.
